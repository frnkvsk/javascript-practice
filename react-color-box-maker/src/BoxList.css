
.BoxList {
  display: flex;
  flex-direction: row;
  /* width: 500px; */
  border: 3px solid green;
}