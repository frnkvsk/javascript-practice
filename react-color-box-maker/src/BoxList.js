/**
 BoxList - Place your state that contains all of the boxes here. 
 This component should render all of the Box components along with the NewBoxForm component
 */
import React, {useState} from 'react';
import ColorBox from './ColorBox';
import { v4 as uuid } from 'uuid';

import NewBoxForm from "./NewBoxForm";

function BoxList() {
  let [boxes, setBoxes] = useState([]);
  const addNewBox = newBox => {
    boxes.push(newBox);
    setBoxes(boxes.slice());
  }
  const hideBox = (id) => {
    boxes = boxes.filter(e => e.id !== id);
    setBoxes(boxes.slice());
  }

  return (
    <div className="BoxList">
      <NewBoxForm addNewBox={addNewBox}/>
      {boxes.map(e => {
        const uid = uuid();
        e.id = uid;
        return <ColorBox 
          key={uid}
          id={uid}
          backgroundColor={e.backgroundColor} 
          width={e.width} 
          height={e.height}
          hideBox={hideBox} />
      }        
      )}
    </div>
  );
}

export default BoxList;