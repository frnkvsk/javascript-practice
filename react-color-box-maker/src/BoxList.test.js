import React from 'react';
import { render } from '@testing-library/react';
import BoxList from './BoxList';

// smoke test
it("Control renders without crashing", () => {
  render(<BoxList />);
});

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});
