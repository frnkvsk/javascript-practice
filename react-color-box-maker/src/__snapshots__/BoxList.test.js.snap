// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form
      autocomplete="off"
      class="makeStyles-root-7"
      novalidate=""
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="form-text-color"
          id="form-text-color-label"
        >
          Background Color
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="form-text-color"
            name="backgroundColor"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Background Color
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="form-text-width"
          id="form-text-width-label"
        >
          Box Width
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="form-text-width"
            name="width"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Box Width
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="form-text-height"
          id="form-text-height-label"
        >
          Box Height
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="form-text-height"
            name="height"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Box Height
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 MuiButton-containedPrimary"
        id="form-btn"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Submit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;
